src/wallet/rpcwallet.cpp

```diff
@@ -2397,6 +2399,40 @@ UniValue lockunspent(const JSONRPCRequest& request)
 
         COutPoint outpt(uint256S(txid), nOutput);
 
+        const bool isLocked = pwallet->IsLockedCoin(outpt.hash, outpt.n);
+
+        if (fUnlock)
+        {
+            if (!isLocked)
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, expected locked output");
+        }
+        else if (!isLocked)
+        {
+            if (pwallet->IsSpent(outpt.hash, outpt.n))
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, expected unspent output");
+
+            const auto it = pwallet->mapWallet.find(outpt.hash);
+            if (it == pwallet->mapWallet.end())
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, unknown transaction");
+
+            const CWalletTx& trans = it->second;
+
+            if (outpt.n >= trans.tx->vout.size())
+                throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, vout index out of bounds");
+
+            if (pwallet->IsMine(trans.tx->vout[outpt.n]) == ISMINE_NO)
```

Comment:
Do we really have to test `IsMine` and all? In the wallet it's just a `std::set<COutPoint>`, so IMO is enough to check:
- if the input is valid output
- if the lock state is compatible with the argument.

But if there is interest in doing these tests, then perform the cheapest first.